releasenotes/notes/ignore-msi-upgrade-version-errors-7eb4b8a78704596b.yaml

```
@@ -0,0 +1,13 @@
+---
+fixes:
+  - |
+    The Agent MSI no longer fails when it is unable to read the
+    version information from [MsiGetProductInfo](https://learn.microsoft.com/en-us/windows/win32/api/msi/nf-msi-msigetproductinfow).
+
+    When upgrading from an Agent version lower than 7.56,
+    if ``MsiGetProductInfo`` fails and the MSI fails and rolls back, then
+    the system may be missing some drivers. If this happens, repair
+    or reinstall the Agent to restore the missing drivers.
+    These cases are usually caused by a corrupted Windows Installer registry.
```

```suggestion

    When upgrading from an Agent version lower than 7.56,
    if ``MsiGetProductInfo`` fails and the MSI fails and rolls back, then
    the system may be missing some drivers. If this happens, repair
    or reinstall the Agent to restore the missing drivers.
    These cases are usually caused by a corrupted Windows Installer registry.
```

**What do you think of this edit:**
When upgrading from an Agent version earlier than 7.56, a failure in MsiGetProductInfo, combined with an MSI rollback, may result in missing system drivers. If this occurs, repairing or reinstalling the Agent will restore the missing drivers. This issue is typically caused by a corrupted Windows Installer registry.

---

https://github.com/DataDog/datadog-agent/pull/38847